### PR TITLE
[#2305] Content editable working on multiline text.

### PR DESCRIPTION
**Pull request by ScottDowne**

### CODE REVIEW

File: templates/assets/editors/editorhelper.js [hunk 1]
```diff
@@ -122,30 +122,43 @@
...
-            trackEvent.update({
-              text: newText && newText !== "" ? newText : trackEvent.popcornOptions.text
-            });
+    global.EditorHelper.contentEditable = function( trackEvent, contentContainers ) {
+      var newText = "",
+          contentContainer;
+
+      var update = function() {
+        newText = "";
+        for ( var i = 0; i < contentContainers.length; i++ ) {
```
Comment (dseif) [inline templates/assets/editors/editorhelper.js]: `for ( var i = 0; i < contentContainers.length; i++ ) {` -> `for ( var i = 0, l = contentContainers.length; i < l; i++ ) {`

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @dseif received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/3b8b22a562612c9/output.txt

File: templates/assets/editors/editorhelper.js [hunk 2]
```diff
@@ -122,30 +122,43 @@
...
-        }, false );
-        contentContainer.addEventListener( "mousedown", function( e ) {
-          if ( !e.shiftKey ) {
-            e.stopPropagation();
-          }
-        }, false );
-        contentContainer.setAttribute( "contenteditable", "true" );
+        }
+        trackEvent.update({
+          text: newText && newText !== "" ? newText : trackEvent.popcornOptions.text
```
Comment (dseif) [inline templates/assets/editors/editorhelper.js]: shouldn't need to do `newText  && newText !== "" ? newText : trackEvent.popcornOptions.text`. Can just do `newText || trackEvent.popcornOptions.text`

File: templates/assets/editors/editorhelper.js [hunk 3]
```diff
@@ -122,30 +122,43 @@
...
+        });
+      };
+
+      for ( var i = 0; i < contentContainers.length; i++ ) {
```
Comment (dseif) [inline templates/assets/editors/editorhelper.js]: same as above mentioned for loop

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/3b8b22a562612c9/output.txt

Total script time: 3.00 mins

File: templates/assets/editors/editorhelper.js [hunk 4]
```diff
@@ -122,30 +122,43 @@
...
+        contentContainer = contentContainers[ i ];
+        if ( contentContainer ) {
+          contentContainer.addEventListener( "blur", function( e ) {
+            update();
+          }, false );
+          contentContainer.addEventListener( "keydown", function( e ) {
+            if ( e.keyCode === 13 ) {
```
Comment (dseif) [inline templates/assets/editors/editorhelper.js]: I'm assuming this is for the enter key? Can you add a comment above just to be clear.

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @dseif received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/d2b7fe5930e3181/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @dseif received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/a5ffa27fdc758f7/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/d2b7fe5930e3181/output.txt

Total script time: 2.80 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/a5ffa27fdc758f7/output.txt

Total script time: 2.39 mins

Server is running on http://pr1054.butyr.org/templates/basic/
Run the Test Suite: http://pr1054.butyr.org/test/

File: templates/assets/editors/editorhelper.js (hunk 2 above)
Comment (ScottDowne) [inline templates/assets/editors/editorhelper.js]: Ah, good call.

That check no longer makes sense.

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/db4d6c1a0169f41/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/94c6906a0eb3a88/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/db4d6c1a0169f41/output.txt

Total script time: 3.00 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/94c6906a0eb3a88/output.txt

Total script time: 3.00 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @jbuck received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/c1c5889c4ac0122/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/c1c5889c4ac0122/output.txt

Total script time: 0.96 mins

Server is running on http://pr1054.butyr.org/templates/basic/
Run the Test Suite: http://pr1054.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/4045c2006d07b4a/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/e280c264ae7ef22/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/4045c2006d07b4a/output.txt

Total script time: 3.00 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/e280c264ae7ef22/output.txt

Total script time: 3.00 mins

**Pull request closed without merging**